services/api_gateway/src/main/java/com/futurewei/alcor/apigateway/filter/KeystoneAuthWebFilter.java

```diff
@@ -0,0 +1,49 @@
+package com.futurewei.alcor.apigateway.filter;
+
+import com.futurewei.alcor.apigateway.utils.KeystoneClient;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.cloud.gateway.support.ServerWebExchangeUtils;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.server.reactive.ServerHttpRequest;
+import org.springframework.stereotype.Component;
+import org.springframework.web.server.ServerWebExchange;
+import org.springframework.web.server.WebFilter;
+import org.springframework.web.server.WebFilterChain;
+import reactor.core.publisher.Mono;
+
+import java.io.IOException;
+
+
+@Component
+public class KeystoneAuthWebFilter implements WebFilter {
```

Comment:
I see most of the payload is feature codes for integration with KeyStone. Can we add it to description accordingly.